### PR TITLE
Upgrade SignalR Function extension dependency versions

### PR DESCRIPTION
**Pull request by Y-Sindo**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 ## 1.9.0-beta.1 (Unreleased)
 
 ### Features Added
-
+* Upgrade `MessagePack` from 1.7.3.7 to 1.9.11 to fix a vulnerability.
```
Comment (heaths) [inline sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/CHANGELOG.md:6]: I wouldn't advertise it so plainly. Just "Upgraded MessagePack for performance and security improvements" is the usual fare.

**Review by heaths — CHANGES_REQUESTED** [review]

File: eng/Packages.Data.props [hunk 2]
```diff
@@ -122,10 +122,11 @@
     <PackageReference Update="Apache.Avro" Version="1.11.0" />
     <PackageReference Update="CloudNative.CloudEvents" Version="2.0.0" />
     <PackageReference Update="CloudNative.CloudEvents.SystemTextJson" Version="2.0.0" />
+    <PackageReference Update="MessagePack" Version="1.9.11" /> <!--Override the dependency version in Microsoft.AspNetCore.SignalR.Protocols.MessagePack to fix a vulnerability.-->
```
[the remark this answers is not in the document]
Comment (weshaggard) [reply eng/Packages.Data.props:125]: Has this dependency been approved?

[the remark this answers is not in the document]
Comment (heaths) [reply eng/Packages.Data.props:125]: It was an indirect dependency already. We were already shipping it, but nuget's default configuration to pick up the earliest shared dependency picks up the vulnerable version. This is so we can reference the newer, fixed one already.

**Review by heaths — APPROVED** [review]

**Comment by heaths** [comment]
This change yields a problem. @Y-Sindo could you look into possible causes and solutions?

```
There was a conflict between "Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" and "Microsoft.Bcl.AsyncInterfaces, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51".
    "Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" was chosen because it was primary and "Microsoft.Bcl.AsyncInterfaces, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" was not.
    References which depend on "Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" [/mnt/vss/_work/1/.nuget/packages/microsoft.bcl.asyncinterfaces/1.1.1/ref/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll].
        /mnt/vss/_work/1/.nuget/packages/microsoft.bcl.asyncinterfaces/1.1.1/ref/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll
          Project file item includes which caused reference "/mnt/vss/_work/1/.nuget/packages/microsoft.bcl.asyncinterfaces/1.1.1/ref/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll".
            /mnt/vss/_work/1/.nuget/packages/microsoft.bcl.asyncinterfaces/1.1.1/ref/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll
    References which depend on "Microsoft.Bcl.AsyncInterfaces, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" [].
        /mnt/vss/_work/1/s/artifacts/bin/Azure.AI.Personalizer/Release/netstandard2.0/Azure.AI.Personalizer.dll
          Project file item includes which caused reference "/mnt/vss/_work/1/s/artifacts/bin/Azure.AI.Personalizer/Release/netstandard2.0/Azure.AI.Personalizer.dll".
            /mnt/vss/_work/1/s/artifacts/bin/Azure.AI.Personalizer/Release/netstandard2.0/Azure.AI.Personalizer.dll
        /mnt/vss/_work/1/s/artifacts/bin/Azure.Identity/Release/netstandard2.0/Azure.Identity.dll
          Project file item includes which caused reference "/mnt/vss/_work/1/s/artifacts/bin/Azure.Identity/Release/netstandard2.0/Azure.Identity.dll".
            /mnt/vss/_work/1/s/artifacts/bin/Azure.Identity/Release/netstandard2.0/Azure.Identity.dll
            /mnt/vss/_work/1/s/artifacts/bin/Azure.Core.TestFramework/Release/net6.0/Azure.Core.TestFramework.dll
##[warning]/opt/hostedtoolcache/dotnet/sdk/6.0.100/Microsoft.Common.CurrentVersion.targets(2304,5): Warning MSB3277: Found conflicts between different versions of "Microsoft.Bcl.AsyncInterfaces" that could not be resolved.
```

**Comment by Y-Sindo** [comment]
> This change yields a problem. @Y-Sindo could you look into possible causes and solutions?
> 
> ```
> There was a conflict between "Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" and "Microsoft.Bcl.AsyncInterfaces, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51".
>     "Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" was chosen because it was primary and "Microsoft.Bcl.AsyncInterfaces, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" was not.
>     References which depend on "Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" [/mnt/vss/_work/1/.nuget/packages/microsoft.bcl.asyncinterfaces/1.1.1/ref/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll].
>         /mnt/vss/_work/1/.nuget/packages/microsoft.bcl.asyncinterfaces/1.1.1/ref/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll
>           Project file item includes which caused reference "/mnt/vss/_work/1/.nuget/packages/microsoft.bcl.asyncinterfaces/1.1.1/ref/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll".
>             /mnt/vss/_work/1/.nuget/packages/microsoft.bcl.asyncinterfaces/1.1.1/ref/netstandard2.1/Microsoft.Bcl.AsyncInterfaces.dll
>     References which depend on "Microsoft.Bcl.AsyncInterfaces, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" [].
>         /mnt/vss/_work/1/s/artifacts/bin/Azure.AI.Personalizer/Release/netstandard2.0/Azure.AI.Personalizer.dll
>           Project file item includes which caused reference "/mnt/vss/_work/1/s/artifacts/bin/Azure.AI.Personalizer/Release/netstandard2.0/Azure.AI.Personalizer.dll".
>             /mnt/vss/_work/1/s/artifacts/bin/Azure.AI.Personalizer/Release/netstandard2.0/Azure.AI.Personalizer.dll
>         /mnt/vss/_work/1/s/artifacts/bin/Azure.Identity/Release/netstandard2.0/Azure.Identity.dll
>           Project file item includes which caused reference "/mnt/vss/_work/1/s/artifacts/bin/Azure.Identity/Release/netstandard2.0/Azure.Identity.dll".
>             /mnt/vss/_work/1/s/artifacts/bin/Azure.Identity/Release/netstandard2.0/Azure.Identity.dll
>             /mnt/vss/_work/1/s/artifacts/bin/Azure.Core.TestFramework/Release/net6.0/Azure.Core.TestFramework.dll
> ##[warning]/opt/hostedtoolcache/dotnet/sdk/6.0.100/Microsoft.Common.CurrentVersion.targets(2304,5): Warning MSB3277: Found conflicts between different versions of "Microsoft.Bcl.AsyncInterfaces" that could not be resolved.
> ```

The packages involved here are not related to this PR. It should be a conflict between `Azure.AI.Personalizer` vs `Azure.Identity`, `Azure.Core.TestFramework`. I saw the same error in other pipelines such as [this one](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1663701&view=logs&j=f1d1026e-546a-5835-36d4-033ab19395ac&t=1e624415-606a-566a-b8e8-5876b361a3ee&l=644).

**Comment by heaths** [comment]
@orenmichaely @tyclintw can you fix whatever is happening with_ sdk/personalizer_? Looks like a conflict of package versions. Note that all `<PackageReferences/>` should not include a `Version` or `OverrideVersion` (by default) and be defined in _eng/Packages.Data.props_ to avoid discrepancies with dependency versions.

**Pull request merged**